sympy/logic/boolalg.py

```diff
@@ -540,6 +540,42 @@ def _eval_derivative(self, x):
         else:
             return S.Zero
 
+    def gateinputcount(self):
+        """
+        Return the total number of inputs for the logic gates realizing the
+        Boolean expression.
+
+        Examples
+        ========
+
+        >>> from sympy.logic import And, Or, Nand, Not
+        >>> from sympy.abc import x, y, z
+        >>> expr = And(x, y)
+        >>> expr.gateinputcount()
+        2
+        >>> Or(expr, z).gateinputcount()
+        4
+
+        Note that Nand is automatically rewritten to Not(And()) so
+        >>> Nand(x, y, z).gateinputcount()
+        4
+        >>> Not(And(x, y, z)).gateinputcount()
+        4
+
+        Also note that a comparison will count as a Boolean vairable.
+        >>> And(x > z, y >= 2).gateinputcount()
+        2
+
+        Returns
+        =======
+
+        int
+            Number of gate inputs
+
+        """
+        return len(self.args) + sum(x.gateinputcount() if isinstance(x, BooleanFunction)
+                                    else 0 for x in self.args)
```

Comment:
I think it's better not to split ternary if/else or generator/list comprehensions over multiple lines:
```python
        subgates = [x for x in self.args if isinstance(x, BooleanFunction)]
        return len(self.args) + sum(g.gateinputcount() for g in subgates)
```